subprojects/base-services/src/test/groovy/org/gradle/internal/service/DefaultServiceRegistryTest.groovy

```diff
@@ -1154,6 +1150,77 @@ class DefaultServiceRegistryTest extends Specification {
         e.message == "Cannot locate factory for objects of type BigDecimal, as TestRegistry has been closed."
     }
 
+    /*
+     * Closing children would imply holding a reference to them. This would
+     * create memory leaks.
+     */
+    def "does not close services from child registries"() {
```

Comment:
I added the tests below, because they would have saved me lots of headaches during this change.